app/uk/gov/hmrc/agentassurance/auth/AuthActions.scala

```diff
@@ -55,16 +56,16 @@ trait AuthActions extends AuthorisedFunctions {
       }
   }
 
-  def AuthorisedWithAgentCode[A](body: AuthorisedRequestWithAgentCode): Action[AnyContent] = Action.async {
+  def AuthorisedWithUserId[A](body: AuthorisedRequestWithUserId): Action[AnyContent] = Action.async {
     implicit request =>
       implicit val hc = fromHeadersAndSession(request.headers, None)
-      authorised(AuthProviders(GovernmentGateway)).retrieve(agentCode) {
-        case Some(agentCode) => body(request)(AgentCode(agentCode))
-        case _ => Future successful NoPermission
+      authorised(AuthProviders(GovernmentGateway)).retrieve(Retrievals.credentials) {
+        case Credentials(providerId, _) => body(request)(providerId)
       } recoverWith {
         case ex: NoActiveSession =>
           Logger.warn("NoActiveSession while trying to access check IR SA endpoint", ex)
           Future.successful(Unauthorized)
+        case _ => Future successful NoPermission
```

Comment:
I think this catchall should be removed as it interferes with the standard error handling in case of upstream 5xx errors. e.g. say auth is down and we get a 503 response from it, we'd return a NoPermission rather than the more appropriate and descriptive 502 that the error handler would return.

So I think we should keep the previous "case _ => Future successful NoPermission" back in the .retrieve{} block.